sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -81,6 +93,16 @@ public HttpConnection(String requestMethod,
         this.url = url;
         this.contentType = contentType;
         this.requestProperties = new HashMap<String, String>();
+        this.requestFilters = new LinkedList<HttpConnectionRequestFilter>();
+        this.responseFilters = new LinkedList<HttpConnectionResponseFilter>();
+    }
+
+    /**
+     * Sets the number of times this request can be retried.
+     * @param numberOfRetries the number of times this request can be retried.
+     */
+    public void setNumberOfRetries(int numberOfRetries){
+        this.numberOfRetries = numberOfRetries;
```

Comment:
Seems like a doc change is fine to me. "Set numberOfRetries before executing the request"